lib/routes/routes.dart

```diff
@@ -365,63 +366,74 @@ class AppRouter extends RouterDelegate<AppRoutes<Object?>>
 
   @override
   Widget build(BuildContext context) {
-    return _AppRouterBuilder(
-      builder: (context) {
-        final pages = <Page<void>>[
-          StackFadePage(
-            key: AppRoutes.initial.uniqueKey,
-            child: const InitialRoute(),
-            transitionSettings: transitionSettings.initial,
-          ),
-        ];
-
-        for (int i = 0; i < _routes.length; i++) {
-          final route = _routes[i];
-          if (route.hasSameLocation(AppRoutes.settings)) {
-            pages.add(StackFadePage(
-              key: AppRoutes.settings.uniqueKey,
-              child: const SettingsRoute(),
-              transitionSettings: transitionSettings.dismissible,
-            ));
-          } else if (route.hasSameLocation(AppRoutes.generalSettings)) {
-            pages.add(StackFadePage(
-              key: AppRoutes.generalSettings.uniqueKey,
-              child: const GeneralSettingsRoute(),
-              transitionSettings: transitionSettings.theme,
-            ));
-          } else if (route.hasSameLocation(AppRoutes.themeSettings)) {
-            pages.add(StackFadePage(
-              key: AppRoutes.themeSettings.uniqueKey,
-              child: const ThemeSettingsRoute(),
-              transitionSettings: transitionSettings.theme,
-            ));
-          } else if (route.hasSameLocation(AppRoutes.licenses)) {
-            pages.add(StackFadePage(
-              key: AppRoutes.licenses.uniqueKey,
-              child: const LicensePage(),
-              transitionSettings: transitionSettings.dismissible,
-            ));
-          } else if (route.hasSameLocation(AppRoutes.dev)) {
-            pages.add(StackFadePage(
-              key: AppRoutes.dev.uniqueKey,
-              child: const DevRoute(),
-              transitionSettings: transitionSettings.dismissible,
-            ));
-          } else if (route.hasSameLocation(AppRoutes.selection)) {
-            pages.add(StackFadePage(
-              key: AppRoutes.selection.uniqueKey,
-              child: SelectionRoute(
-                selectionArguments: (route as AppRoutes<SelectionArguments>).arguments!,
-              ),
-              transitionSettings: transitionSettings.greyDismissible,
-            ));
-          }
+    return Consumer(
+      builder: (context, ref, child) {
+        final mediaQuery = MediaQuery.of(context);
+        final textScaleFactor = ref.watch(textScaleFactorStateNotifierProvider.select((value) => value));
+        if (mediaQuery.size.isEmpty) {
+          return const SizedBox.shrink(); // Don't render the app if we are started in the background.
         }
```

Comment:
When started in the background via a media play button, the `AudioService` will start the app without a window and `mediaQuery.size` will be empty. If we don't do an early return here we would do a lot of unnecessary work and also throw some divide by zero errors because of the `(0, 0)` window size.

When the app is started in the foreground the `MediaQuery` is updated and the app is drawn as normal.